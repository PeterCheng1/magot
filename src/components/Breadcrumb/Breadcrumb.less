@import '../../variable.less';

.mgt-breadcrumb {
  position: relative;
}
