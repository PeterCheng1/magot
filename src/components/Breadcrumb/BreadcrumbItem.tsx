import React from 'react';

import * as component from '../component';

export interface BreadcrumbItemProps extends component.ComponentBase {}

const defaultProps: Partial<BreadcrumbItemProps> = {};

function BreadcrumbItem(props: BreadcrumbItemProps) {
  const cls = component.getComponentClasses('breadcrumb-item', props);
  return <div className={cls} />;
}

BreadcrumbItem.defaultProps = defaultProps;

export default BreadcrumbItem;
