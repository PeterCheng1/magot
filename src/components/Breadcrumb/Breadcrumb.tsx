import React from 'react';

import './Breadcrumb.less';

import * as component from '../component';

export interface BreadcrumbProps extends component.ComponentBase {}

const defaultProps: Partial<BreadcrumbProps> = {};

function Breadcrumb(props: BreadcrumbProps) {
  const cls = component.getComponentClasses('breadcrumb', props);
  return <div className={cls} />;
}

Breadcrumb.defaultProps = defaultProps;

export default Breadcrumb;
