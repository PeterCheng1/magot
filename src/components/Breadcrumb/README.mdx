---
name: Breadcrumb
route: /components/Breadcrumb
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Breadcrumb from './Breadcrumb';

# Breadcrumb

## Usage

## props

<PropsTable of={Breadcrumb} />
