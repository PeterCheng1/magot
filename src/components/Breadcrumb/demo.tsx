import React from 'react';
import { Link } from 'react-router-dom';
import Breadcrumb from './Breadcrumb';

function BreadcrumbDemo() {
  return (
    <>
      <Link to="/" className="demo-goback">
        返回
      </Link>
      <div className="demo-box">
        <Breadcrumb>
          <Breadcrumb.Item>Home</Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">User</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">Template</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">Editor</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>Workbench</Breadcrumb.Item>
        </Breadcrumb>
      </div>
      <div className="demo-box">
        <Breadcrumb>
          <Breadcrumb.Item icon="home" />
          <Breadcrumb.Item icon="user">
            <a href="">User</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item icon="store">
            <a href="">Template</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">Editor</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>Workbench</Breadcrumb.Item>
        </Breadcrumb>
      </div>
      <div className="demo-box">
        <Breadcrumb separator="&gt;">
          <Breadcrumb.Item>Home</Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">User</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">Template</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="">Editor</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>Workbench</Breadcrumb.Item>
        </Breadcrumb>
      </div>
    </>
  );
}

export default BreadcrumbDemo;
