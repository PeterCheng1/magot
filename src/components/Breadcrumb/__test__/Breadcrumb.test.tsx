import React from 'react';
import { render } from 'enzyme';
import Breadcrumb from '../Breadcrumb';
import { injectStyleTestSuites } from '../../../../test/shared.test';

describe('Breadcrumb', () => {
  it('should renders dom correctly', () => {
    const wrapper = render(
      <Breadcrumb>
        <Breadcrumb.Item>Home</Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">User</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Template</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Editor</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>Workbench</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders empty children', () => {
    const wrapper = render(<Breadcrumb />);
    expect(wrapper.children()).toHaveLength(0);
  });

  it('should renders breadcrumb items', () => {
    const wrapper = render(
      <Breadcrumb>
        <Breadcrumb.Item>Home</Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">User</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Template</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Editor</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>Workbench</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(wrapper.find('.mgt-breadcrumb-item')).toHaveLength(5);
  });

  it('should renders a breadcrumb item', () => {
    let wrapper = render(<Breadcrumb.Item>Home</Breadcrumb.Item>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('Home');
    expect(wrapper.find('.mgt-breadcrumb-separator')).toHaveLength(1);

    wrapper = render(
      <Breadcrumb.Item>
        <a href="">User</a>
      </Breadcrumb.Item>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('a')).toHaveLength(1);
    expect(wrapper.find('.mgt-breadcrumb-separator')).toHaveLength(1);
  });

  it('should renders `separator` prop when passed in', () => {
    const wrapper = render(
      <Breadcrumb separator="&gt;">
        <Breadcrumb.Item>Home</Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">User</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Template</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="">Editor</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>Workbench</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-breadcrumb-separator').text()).toEqual('>>>>>');
  });

  injectStyleTestSuites(Breadcrumb);
});
