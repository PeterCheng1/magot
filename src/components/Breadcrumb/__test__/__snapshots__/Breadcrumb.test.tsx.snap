// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should renders \`separator\` prop when passed in 1`] = `
<ul
  class="mgt-breadcrumb"
>
  <li
    class="mgt-breadcrumb-item"
  >
    Home
    <span
      class="mgt-breadcrumb-separator"
    >
      &gt;
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      User
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      &gt;
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      Template
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      &gt;
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      Editor
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      &gt;
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    Workbench
    <span
      class="mgt-breadcrumb-separator"
    >
      &gt;
    </span>
  </li>
</ul>
`;

exports[`Breadcrumb should renders a breadcrumb item 1`] = `
<li
  class="mgt-breadcrumb-item"
>
  Home
  <span
    class="mgt-breadcrumb-separator"
  />
</li>
`;

exports[`Breadcrumb should renders a breadcrumb item 2`] = `
<li
  class="mgt-breadcrumb-item"
>
  <a
    href=""
  >
    User
  </a>
  <span
    class="mgt-breadcrumb-separator"
  />
</li>
`;

exports[`Breadcrumb should renders custom className 1`] = `
<ul
  class="mgt-breadcrumb jest-test-class"
/>
`;

exports[`Breadcrumb should renders custom styles 1`] = `
<ul
  class="mgt-breadcrumb"
  style="color:#333;background-color:#f0f0f0"
/>
`;

exports[`Breadcrumb should renders dom correctly 1`] = `
<ul
  class="mgt-breadcrumb"
>
  <li
    class="mgt-breadcrumb-item"
  >
    Home
    <span
      class="mgt-breadcrumb-separator"
    >
      /
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      User
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      /
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      Template
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      /
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    <a
      href=""
    >
      Editor
    </a>
    <span
      class="mgt-breadcrumb-separator"
    >
      /
    </span>
  </li>
  <li
    class="mgt-breadcrumb-item"
  >
    Workbench
    <span
      class="mgt-breadcrumb-separator"
    >
      /
    </span>
  </li>
</ul>
`;
